lua/entities/gmod_artron_inhibitor/init.lua

```diff
@@ -0,0 +1,214 @@
+end
+
+function ENT:Repair(repair)
+    if self.EntHealth >= 0 then
+        local hp = self.EntHealth + repair
+        
+        if self.Broken == true and hp >= 0 then
+            self:SetColor(Color(255, 255, 255, self:GetColor().a))
+            self.Broken = false
+        end
+
+        self.EntHealth = math.Clamp(hp,0,self.EntMaxHealth)
+        self:TriggerWire("Health",self.EntHealth)
+    end
+end
+
+function ENT:TriggerInput(iname, value)
+    if iname == "Activate" then
+        if self.Broken then return end
+
+        if value >= 1 and self:GetEnabled() == false then
+            self:TurnOn(true)
+        else
+            self:TurnOn(false)
+        end
+    elseif iname == "Radius" then
+        self:SetRadius(value)
+    end
+end
+
+function ENT:TurnOn(active)
+    if self.Broken then return end
+
+    local phys = self:GetPhysicsObject()
+
+    if active and self.FlyTime == nil then
+        self:SetColor(Color(255, 166, 0, self:GetColor().a))
+        phys:SetAngleVelocity(Vector(20, -20, -10))
+        phys:EnableGravity(false)
+        phys:SetVelocity(Vector(0, 0, 15))
+
+        self:TriggerWire("Active",0.5)
+
+        sound.Play("drmatt/tardis/seq_ok.wav", self:GetPos())
+
+        if IsValid(self.LastActivator) then
+            TARDIS:Message(self.LastActivator, "ArtronInhibitor.Starting")
+        end
+
+        self.FlyTime = CurTime()
+        self.ArtronTick = CurTime()
+    elseif self.On == true then
+        self.FlyTime = nil
+        self.ArtronTick = nil
+        
+        self.On = false
+        self:SetEnabled(false)
+        self:TriggerWire("Active",0)
+        self:SetColor(Color(255, 255, 255, self:GetColor().a))
+
+        phys:EnableGravity(true)
+
+        sound.Play("drmatt/tardis/seq_bad.wav", self:GetPos())
+
+        if IsValid(self.LastActivator) then
+            TARDIS:Message(self.LastActivator, "ArtronInhibitor.Disabled")
+        end
+
+        local effect_data = EffectData()
+        effect_data:SetOrigin(self:GetPos())
+        util.Effect("cball_explode", effect_data)
+    end
+end
+
+function ENT:Use(ply)
+    if self.Broken then return end
+
+    self.LastActivator = ply
+
+    if self:GetEnabled() then
+        self:TurnOn(false)
+    else
+        self:TurnOn(true)
+    end
+end
+
+function ENT:Think()
+    if self.Broken then return end
+
+    if self.On then
+        local phys = self:GetPhysicsObject()
+        if phys:IsMotionEnabled() then
+            phys:SetAngleVelocity(Vector(180, -200, -190))
+            phys:AddVelocity(-0.2 * phys:GetVelocity())
+        end
+    end
+
+    if self.FlyTime ~= nil and CurTime() - self.FlyTime > 3 then
+        self.FlyTime = nil
+        self.On = true
+        self:SetEnabled(true)
+
+        TARDIS:Message(self.LastActivator, "ArtronInhibitor.Enabled")
+        self.LastActivator = nil
+
+        self:GetPhysicsObject():SetVelocity(Vector(0, 0, 0))
+        self:SetColor(Color(164, 90, 250, self:GetColor().a))
```

Comment:
It would make sense to have this be a different colour to the TDG

Reply:
started color is now orange (the startup color of the tdg because annoyingly its a good color for artron energy)
startup color is blue